http/RemoteResource.cc

```diff
@@ -421,7 +344,119 @@ namespace http {
             throw;
         }
 
-    }
+    } //end RemoteResource::retrieveResource()
+
+    void RemoteResource::update_file_and_headers(const string &template_key, const string &replace_value){
+
+        // Get a pointer to the singleton cache instance for this process.
+        HttpCache *cache = HttpCache::get_instance();
+        if (!cache) {
+            ostringstream oss;
+            oss << prolog << "FAILED to get local cache. ";
+            oss << "Unable to proceed with request for " << this->d_remoteResourceUrl;
+            oss << " The server MUST have a valid HTTP cache configuration to operate." << endl;
+            BESDEBUG(MODULE, oss.str());
+            throw BESInternalError(oss.str(), __FILE__, __LINE__);
+        }
+
+        // Write the remote resource to the cache file.
+        try {
+            writeResourceToFile(d_fd);
+        }
+        catch (...) {
+            // If things went south then we need to dump the file because we'll end up with an empty/bogus file clogging the cache
+            unlink(d_resourceCacheFileName.c_str());
+            throw;
+        }
+
+        //  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+        // If we are filtering the response (for example to inject data URL into a dmr++ file),
+        // The file is locked and we have the information required to make the substitution.
+        // This is controlled by:
+        //  - The template_key string must not be empty.
+        if(!template_key.empty()){
+            unsigned int count = filter_retrieved_resource(template_key, replace_value);
+            BESDEBUG(MODULE, prolog << "Replaced " << count <<
+                                    " instance(s) of template(" <<
+                                    template_key << ") with " << replace_value << " in cached RemoteResource" << endl);
+        }
+
+        //  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+        // I think right here is where I would be able to cache the data type/response headers. While I have
+        // the exclusive lock I could open another cache file for metadata and write to it.
+        {
+            // FIXME THIS IS WHERE WE NEED TO USE NGAP HASH CACHE FILE NAMES NOT FLC NAMES.
```

Comment:
Does this still need to be done? Also, rephrase the 'I think ...' comment unless there's still some question.

Reply:
These comments are cruft and could be removed.

Reply:
removed